lib/view_models/page_view_model/join_organization_viewModel.dart

```diff
@@ -0,0 +1,358 @@
+                    _loadingIndex = index;
+                    _isLoaderActive = true;
+                    notifyListeners();
+                    Navigator.of(context).pop();
+                    if (isPublic == 'true') {
+                      await joinPublicOrg(context, orgName).whenComplete(() {
+                        _loadingIndex = -1;
+                        _isLoaderActive = false;
+                      });
+                    } else if (isPublic == 'false') {
+                      await joinPrivateOrg(context).whenComplete(() {
+                        _loadingIndex = -1;
+                        _isLoaderActive = false;
+                        notifyListeners();
+                      });
+                    }
+                  },
+                  child: const Text("Yes"),
+                )
+              ],
+            );
+          });
+    }
+  }
+
+  successToast(String msg) {
+    final Widget toast = Container(
+      padding: EdgeInsets.symmetric(
+          horizontal: SizeConfig.safeBlockHorizontal * 5,
+          vertical: SizeConfig.safeBlockVertical * 1.5),
+      decoration: BoxDecoration(
+        borderRadius: BorderRadius.circular(25.0),
+        color: Colors.green,
+      ),
+      child: Row(
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          Text(msg),
+        ],
+      ),
+    );
+
+    fToast.showToast(
+      child: toast,
+      gravity: ToastGravity.BOTTOM,
+      toastDuration: const Duration(seconds: 3),
+    );
+  }
+
+  exceptionToast(String msg) {
+    final Widget toast = Container(
+      padding: EdgeInsets.symmetric(
+          horizontal: SizeConfig.safeBlockHorizontal * 6,
+          vertical: SizeConfig.safeBlockVertical * 1.75),
+      decoration: BoxDecoration(
+        borderRadius: BorderRadius.circular(25.0),
+        color: Colors.red,
+      ),
+      child: Text(msg),
+    );
+
+    fToast.showToast(
+      child: toast,
+      gravity: ToastGravity.BOTTOM,
+      toastDuration: const Duration(seconds: 3),
+    );
+  }
+
+  void searchOrgName(String orgName) {
+    //it is the search bar to search the organization
+    _filteredOrgInfo.clear();
+    if (orgName.isNotEmpty) {
+      for (int i = 0; i < organizationInfo.length; i++) {
+        final String name = organizationInfo[i]['name'].toString();
+        if (name.toLowerCase().contains(orgName.toLowerCase())) {
+          _filteredOrgInfo.add(organizationInfo[i]);
+          notifyListeners();
+        }
+      }
+    } else {
+      _filteredOrgInfo.add(organizationInfo);
+      notifyListeners();
+    }
+  }
+
+  // Function for getting the current user id.
+  Future<void> getCurrentUserId() async {
+    _currentUserId = await _pref.getUserId();
+    notifyListeners();
+  }
+
+  Future fetchOrg() async {
+    getCurrentUserId();
+    //function to fetch the org from the server
+    final GraphQLClient _client = graphQLConfiguration.authClient();
+
+    final QueryResult result = await _client
```

Comment:
rename to more meaning variable. is this fetching all the organizations?

Reply:
Renamed in the latest commit.😊